pagopa/pn-frontend
# feat/PN-13016: Add taxonomy error code (PA)

## Short description
This PR adds a new PN_GENERIC_INVALIDPARAMETER_TAXONOMYCODE to AppError codes and shows a custom error message when the response contains this error code.

## How to test
Login into PA and go to "send new notification", intercept the new notification api request (https://webapi.dev.notifichedigitali.it/bff/v1/notifications/sent method: POST) and mock the response to return a 4xx error code and the following body:
`
{
    "type": "GENERIC_ERROR",
    "status": 400,
    "title": "Handled error",
    "detail": "See logs for details in PN-BFF",
    "traceId": "Self=1-6728cffd-6cf7f7631eb6fcee00d1c6f3;Root=1-6728cffd-7c1be7ab01191a4a348305e8;Parent=6a1bfee56dae4d55;Sampled=0",
    "timestamp": "2024-11-04T13:45:34.197Z",
    "errors": [
        {
            "code": "PN_GENERIC_INVALIDPARAMETER_TAXONOMYCODE",
            "element": null,
            "detail": "none"
        }
    ]
}
`

Go through the entire 'new notification' wizard and verify that the desired error message is shown at the end of the process.
